app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -225,98 +253,165 @@ fun AssociationRecruitment(context: Context) {
-              .padding(horizontal = 23.dp)
-              .width(366.dp)
-              .height(40.dp)
-              .background(Color.LightGray, RoundedCornerShape(12.dp))
-              .padding(vertical = 2.dp, horizontal = 3.dp)
-              .clickable {
-                scope!!.launch {
-                  testSnackbar!!.showSnackbar(
-                      message = DEBUG_MESSAGE, duration = SnackbarDuration.Short)
-                }
-              },
-  ) {
-    Row(
-        horizontalArrangement = Arrangement.spacedBy(115.dp, Alignment.Start),
-        verticalAlignment = Alignment.CenterVertically,
+  userList.forEach { user ->
+    Box(
+        modifier =
+            Modifier.padding(horizontal = 23.dp)
+                .width(366.dp)
+                .height(40.dp)
+                .background(Color.LightGray, RoundedCornerShape(12.dp))
+                .padding(vertical = 2.dp, horizontal = 3.dp)
+                .clickable {
+                  scope!!.launch {
+                    testSnackbar!!.showSnackbar(
+                        message = DEBUG_MESSAGE, duration = SnackbarDuration.Short)
+                  }
+                },
     ) {
-      Icon(Icons.Filled.Person, contentDescription = "user's profile picture", Modifier.size(36.dp))
-      Text(text = "<Casey Rue>", style = AppTypography.headlineSmall)
+      Row(
+          horizontalArrangement = Arrangement.spacedBy(115.dp, Alignment.Start),
+          verticalAlignment = Alignment.CenterVertically,
+      ) {
+        Icon(
+            Icons.Filled.Person,
+            contentDescription = "user's profile picture",
+            Modifier.size(36.dp))
+        Text(text = user.firstName + " " + user.lastName, style = AppTypography.headlineSmall)
+      }
     }
   }
 }
 
+/**
+ * Component that display all the events of the association in a card format, like in the home
+ * screen.
+ *
+ * @param association (Association) : The association currently displayed
+ * @param associationViewModel (AssociationViewModel) : The associations view model
+ * @param context (Context) : The context of the screen
+ */
 @Composable
-fun AssociationProfileEvents(context: Context) {
-  Column(
-      modifier = Modifier.padding(horizontal = 28.dp),
-      horizontalAlignment = Alignment.CenterHorizontally) {
-        Box(modifier = Modifier.testTag("AssociationEventCard")) {
-          EventCard(
-              event =
-                  Event(
-                      organisers = MockReferenceList(),
-                      taggedAssociations = MockReferenceList())) {}
+private fun AssociationProfileEvents(
+    association: Association,
+    associationViewModel: AssociationViewModel,
+    context: Context
+) {
+  var isSeeMoreClicked by remember { mutableStateOf(false) }
+  var events = emptyList<Event>()
+  associationViewModel.getEventsForAssociation(association) { fetchedEvents ->
+    events = fetchedEvents
+  }
+  if (events.isEmpty()) {
```

Comment:
I couldn't test because as of now no association has any events, but I think that events.isEmpty() will always return true because it does not wait for getEventsForAssociation(association) to finish. Tell me if I am wrong but in my understanding the condition and following code should be in the onSuccess lambda of getEventsForAssociation

Reply:
Maybe this will never happend, but in case this will, the app will not crash and it will just display some text. That's why I added this. Tell me if it should be deleted even if we consider this